device/thunder_ripple_sdk/src/processors/thunder_device_info.rs

```diff
@@ -820,7 +823,26 @@ impl ThunderDeviceInfoRequestProcessor {
         .is_ok()
     }
 
-    async fn get_video_resolution(state: &CachedState) -> Vec<i32> {
+    async fn get_current_resolution(state: &CachedState) -> Result<Vec<i32>, ()> {
+        let response = state
+            .get_thunder_client()
+            .call(DeviceCallRequest {
+                method: ThunderPlugin::DisplaySettings.method("getCurrentResolution"),
+                params: None,
+            })
+            .await;
+        if response.message.get("success").is_none()
+            || !response.message["success"].as_bool().unwrap_or_default()
+        {
+            error!("{}", response.message);
+            return Err(());
+        }
+        info!("{}", response.message);
+        let resol = response.message["resolution"].as_str().unwrap();
```

Comment:
What if the response message doesn't have `resolution`  field ?  
`as_str() ` will return `None` , leads to panic!